cli/src/lib.rs

```diff
@@ -816,58 +833,81 @@ fn docker_build(
         return Err(anyhow!("Failed to build program"));
     }
 
+    let result = docker_prep(container_name, build_config).and_then(|_| {
+        let cfg_parent = cfg.path().parent().unwrap();
+        docker_build_bpf(
+            container_name,
+            cargo_toml.as_path(),
+            cfg_parent,
+            target_dir.as_path(),
+            binary_name,
+            stdout,
+            stderr,
+            cargo_args,
+        )
+    });
+
+    // Cleanup regardless of errors
+    docker_cleanup(container_name, target_dir.as_path())?;
+
+    // Done.
+    result
+}
+
+fn docker_prep(container_name: &str, build_config: &BuildConfig) -> Result<()> {
     // Set the solana version in the container, if given. Otherwise use the
     // default.
-    if let Some(solana_version) = solana_version {
+    if build_config.bootstrap {
+        // Install build requirements
+        docker_exec(container_name, &["apt", "update"])?;
+        docker_exec(
+            container_name,
+            &["apt", "install", "-y", "curl", "build-essential"],
+        )?;
```

Comment:
> Instead of using the default anchor docker image, allow setting the image name, along with another flag for bootstrapping the image with all requirements for building Solana programs: the Solana CLI, Rust, and curl.

Looks like this will work only with debian-based images, not sure how to fix :thinking:
Overall LGTM.

Reply:
That's a very good point.  In the meantime, we can do 2 things:
* change the flag name to `--bootstrap-debian`
* add info that the flag only works for debian-based images
what do you think?

Reply:
Both proposals look good :+1: 
I think it still would be great to rename `--bootstrap` to `--bootstrap-debian` so in the future we would be able to add `--bootstrap` as a path to the bootstrap script?

Reply:
ah nice, great idea! I switched it to an enum, which means we could eventually add a "catchall" at the end which interprets the string as a script path, and `debian` essentially runs an included script.  Sort of similar to how passing `solana config set -u mainnet-beta` expands out to `solana config set -u https://api.mainnet-beta.solana.com`.  Let me know how it looks

Reply:
enum looks good! :+1: